File: wrappers/javascript/anoncreds-shared/src/api/PresentationRequest.ts
import type { JsonObject } from '../types'

import { AnoncredsObject } from '../AnoncredsObject'
import { anoncreds } from '../register'

export class PresentationRequest extends AnoncredsObject {
  public static fromJson(json: JsonObject) {
    return new PresentationRequest(anoncreds.presentationRequestFromJson({ json: JSON.stringify(json) }).handle)
  }
}

File: wrappers/dotnet/lib/Requests/PresentationRequest.cs
using AnonCredsNet.Models;

namespace AnonCredsNet.Requests;

public sealed class PresentationRequest : AnonCredsObject
{
    internal PresentationRequest(long handle)
        : base(handle) { }

    public static PresentationRequest FromJson(string json) => FromJson<PresentationRequest>(json);
}
